tests/testthat/test-release.R

```diff
@@ -199,3 +199,38 @@ test_that("get_release_data() works for new-style CRAN-RELEASE", {
   expect_equal(res$SHA, HEAD)
   expect_equal(path_file(res$file), "CRAN-SUBMISSION")
 })
+
+test_that("cran_version() is robust to unset CRAN mirror with unreleased pkg (#1857)", {
+  create_local_package()
+
+  cran_repos <- list(
+    c(CRAN = "https://cloud.r-project.org"),
+    c(CRAN = "@CRAN@"),
+    c(CRAN = NA),
+    NULL
+  )
+
+  lapply(cran_repos, function(x) {
```

Comment:
I have always found test failures inside this sort of a loop to be reported in a suboptimal way, although I haven't explored that space recently. But I'd be totally fine with doing this the "big and dumb" way, i.e. separate expectations, each inside a `withr::with_options()`.